lib/xml_utils.pm

```diff
@@ -87,7 +97,7 @@ sub find_nodes {
     my (%args) = @_;
     my $nodeset = $args{xpc}->findnodes($args{xpath});
     for my $node ($nodeset->get_nodelist) {
-        print $node->to_literal;
+        diag $node->to_literal;
```

Comment:
why diag here?

Reply:
I'm not sure if helped me to debug something in some point, I cannot recall, but in general we should use `diag` instead of `print` because it is more reliable, it is part of os-autoinst.

Reply:
i didnt know that there is diag implementation in os-autoinst. i thought it was in the test module. from the documentation "Prints a diagnostic message which is guaranteed not to interfere with test output. Like print @diagnostic_message is simply concatenated together." As i am not expert in perl i am leaving it for others to decide(or you). out of curiosity, pm a link of its output if you want

Reply:
Now I'm thinking, I have added another `diag` when searching by value, so now we can see printed in the logs all values are evaluated, not only when searching by node like in [here](http://rivera-workstation.suse.cz/tests/687/file/autoinst-log.txt) when you search `[2019-12-03T15:25:16.141 CET] [debug] gpt` you can see them. Thanks for taking a look, I would have not noticed that. I think before was not printed for some reason, also this thing is running in the worker. Let's leave it like that, seems useful.